packages/calcite-components/src/components/modal/modal.tsx

```diff
@@ -87,8 +87,7 @@ export class Modal
   @Prop({ mutable: true, reflect: true }) open = false;
 
   /** Passes a function to run before the component closes. */
-  @Prop()
-  beforeClose: (el: HTMLElement) => Promise<void> = () => Promise.resolve();
+  @Prop() beforeClose: (el: HTMLCalciteModalElement) => Promise<void>;
```

Comment:
For posterity, this is removing the default value, but we should be OK as it's [documented as `undefined`](https://developers.arcgis.com/calcite-design-system/components/modal/#api-reference-properties-beforeClose).